### PR TITLE
[fix](Export) Fix the problem of exporting stuck

### PR DESCRIPTION
**Pull request by BePPPower**
### What problem does this PR solve?
Problem Summary:

The `disruptor` will be witing when the ringbuffer does not have enough capacity. At the same time, `addExportJobAndRegisterTask` will not release the lock of `ExportMgr`. This prevents other methods from obtaining the lock.

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [x] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [x] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run cloud_p0

File: fe/fe-core/src/main/java/org/apache/doris/scheduler/disruptor/TaskDisruptor.java [hunk 1]
```diff
@@ -119,15 +120,15 @@ public void tryPublish(Long jobId, Long taskId, TaskType taskType) {
...
      * @param taskId task id
      */
-    public void tryPublishTask(Long taskId) {
+    public void tryPublishTask(Long taskId) throws JobException {
         if (isClosed) {
             log.info("tryPublish failed, disruptor is closed, taskId: {}", taskId);
             return;
         }
-        try {
+        if (disruptor.getRingBuffer().hasAvailableCapacity(2)) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/scheduler/disruptor/TaskDisruptor.java:128]: Add comment to explain what does `2` mean.

File: fe/fe-core/src/main/java/org/apache/doris/load/ExportMgr.java [hunk 2]
```diff
@@ -108,26 +108,24 @@ public void addExportJobAndRegisterTask(ExportJob job) throws Exception {
...
+        // delete existing files
+        if (Config.enable_delete_existing_files && Boolean.parseBoolean(job.getDeleteExistingFiles())) {
+            if (job.getBrokerDesc() == null) {
+                throw new AnalysisException("Local file system does not support delete existing files");
+            }
+            String fullPath = job.getExportPath();
+            BrokerUtil.deleteDirectoryWithFileSystem(fullPath.substring(0, fullPath.lastIndexOf('/') + 1),
+                    job.getBrokerDesc());
+        }
+        Env.getCurrentEnv().getEditLog().logExportCreate(job);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/load/ExportMgr.java:123]: I think `Env.getCurrentEnv().getEditLog().logExportCreate(job);` should be done within the lock.

1. unprotectAddJob(job);
2. Env.getCurrentEnv().getEditLog().logExportCreate(job);
3. delete existing files
4. addMemoryTask

**Comment by BePPPower** [comment]
run buildall

**Review by morningman — APPROVED** [review]

**Review by hubgeter — APPROVED** [review]

**Pull request merged**